### PR TITLE
CI: Fold and merge tables in parsing tests results summary.

### PR DESCRIPTION
**Pull request by mglb**
When everything passed, the list of tests is initially folded.

[Live demo](https://github.com/antmicro/yosys-systemverilog/actions/runs/4258279850)

Screen Shot 2023-02-27 at 13 04 22

---

When there are failures, their list is initially visible. The separate list of passed tests is initially folded. Note: failures on the screenshot are made up and were artificially injected into test outputs just to show and test how it would look.

[Live demo](https://github.com/antmicro/yosys-systemverilog/actions/runs/4258278521)

Screen Shot 2023-02-27 at 13 04 36

---

Up until now there were two tables. Both with the same tests, but with results for different tool.
Now there's just one table, with results for both tools in separate columns.

Screen Shot 2023-02-27 at 13 05 29

### CODE REVIEW

**Review by wsipak — CHANGES_REQUESTED** [review]

**Review by kamilrakoczy — APPROVED** [review]

File: .github/scripts/generate_summary.sh [hunk 1]
```diff
@@ -39,34 +44,80 @@ else
...
-    echo "$LINE" >> $GITHUB_STEP_SUMMARY
-done < $SYSTEMVERILOG_FAILED_FILE
-while IFS= read -r LINE; do
-    echo "$LINE" >> $GITHUB_STEP_SUMMARY
-done < $SYSTEMVERILOG_PASSED_FILE
-echo "" >> $GITHUB_STEP_SUMMARY
+declare -a results_00=()
+declare -a results_10=()
+declare -a results_01=()
+declare -a results_11=()
```
[the remark this answers is not in the document]
Comment (mglb) [reply .github/scripts/generate_summary.sh:77]: They represent success or failure, yes. However, they also represent an actual value read from status files.
As you can see in the `case` below, it checks `${uhdm_results[$test_name]:-0}${systemverilog_results[$test_name]:-0}`, which expands to one of: `00`, `01`, `10`, `11`. So e.g. `results_00` contains literally "test names with results, respectivelly, `0` and `0`.
Another advantage of those names is that it is harder to mess up case value and array name when you have code like `11) results_11+=...`. It would be even better with `0` instead of `[^1]`.
You can also think about digits in those names just as "status in the first column" and "status in the second column", without really linking them to actual names. However, this would work better if the column names were stored in an array instead of being hardcoded.

TL;DR: Would comment above the names be sufficient? If not, I'll change the names, as the suggestion is generally OK.

Note:
The case actually checks for anything except `1` instead of zero, just in case something weird ends up in the file instead of `0`. I think such thing happening has a very low probability. I'm willing to change `[^1]` to `0` if you would like so.

[the remark this answers is not in the document]
Comment (wsipak) [reply .github/scripts/generate_summary.sh:77]: Alright, I can see these names actually make sense. I think we would benefit from having a comment clarifying what these names mean, and that should be sufficient. Thanks!
